### PR TITLE
feat(min_charge): Ability to create prorated true up fees

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  https://getlago.canny.io/feature-requests/p/define-minimum-fees

## Context

We want to allow users to set spendings minimum on usage-based charges.

## Description

The goal of this PR is to be able to create prorated true up fees.

### CODE REVIEW

File: spec/scenarios/terminate_pay_in_advance_spec.rb [hunk 1]
```diff
@@ -236,4 +236,81 @@
...
+
+        expect {
+          terminate_subscription(subscription)
+        }.to change { subscription.reload.status }.from('active').to('terminated')
+          .and change { subscription.invoices.count }.from(1).to(2)
+          .and change { sub_invoice.reload.credit_notes.count }.from(0).to(1)
+
+        term_invoice = subscription.invoices.order(sequential_id: :desc).first
+
+
```
Comment (jdenquin) [inline spec/scenarios/terminate_pay_in_advance_spec.rb:285]: Layout/EmptyLines: Extra blank line detected. (https://rubystyle.guide#two-or-more-empty-lines)

File: spec/services/fees/create_true_up_service_spec.rb [hunk 2]
```diff
@@ -55,5 +55,33 @@
...
+
+          expect(result.true_up_fee).to have_attributes(
+            amount_cents: 283, # (1000 / 31.0 * 15) - 200
+          )
+        end
+      end
+    end
   end
-end
+end
```
Comment (jdenquin) [inline spec/services/fees/create_true_up_service_spec.rb:87]: Layout/TrailingEmptyLines: Final newline missing. (https://rubystyle.guide#newline-eof)

File: app/services/subscriptions/dates/weekly_service.rb [hunk 3]
```diff
@@ -5,8 +5,6 @@ module Dates
     class WeeklyService < Subscriptions::DatesService
       WEEK_DURATION = 7
 
-      private
-
       def compute_base_date
```
Comment (vincent-pochet) [inline app/services/subscriptions/dates/weekly_service.rb:8]: Is there a reason for removing this private keyword?

The idea was to discourage the usage of these internal method from the outside

**Review by vincent-pochet — APPROVED** [review]

File: app/services/subscriptions/dates/weekly_service.rb (hunk 3 above)
Comment (rsempe) [reply app/services/subscriptions/dates/weekly_service.rb:8]: Well - I'm calling `compute_charges_duration` and `compute_charges_from_date` in `CreateTrueUpService` to calculate the day price... That's why 😬

Comment (rsempe) [reply app/services/subscriptions/dates/weekly_service.rb]: Finally updated 👍

**Pull request merged**